R/restopt_component.R

```diff
@@ -0,0 +1,30 @@
+#' @include internal.R
+NULL
+
+#' Restoration problem component
+#'
+#' Create a component to add to a restoration optimization problem.
+#'
+#' @param name `character` Name of the component.
+#'
+#' @param java `character` Name Java method for adding component.
+#'
+#' @param class `character` Vector of classes to identify the component.
+#'
+#' @param data `list` Object containing parameters for adding the component.
+#'  Defaults to an empty list.
+#'
+#' @param args `character` Vector used to map the names of the elements in
```

Comment:
This won't help, as arguments in Java methods are not named. I'd rather avoid trying to design a generic mapping with the Java API, as I think it will just overcomplicate things in our case (Java calls will only consist of one line per constraint/objective, and I do not think we will exceed 10 constraints).

Reply:
Ah - thank you very much for catching! You're right, I mistakenly assumed we could use named arguments to map with the Java API. So does that mean that as long as the arguments are specified in the right order, then it should work? If so, then we could just make sure that the elements in `data` are in the right order for the Java code - and that should work? Is that right?

Reply:
Yes in Java methods/functions arguments are ordered, we just need to call the corresponding java function in the correct order.

Reply:
Ah ok - excellent!